sycl/include/CL/sycl/range.hpp

```diff
@@ -37,6 +37,12 @@ class range : public detail::array<dimensions> {
   range(typename std::enable_if<(N == 3), size_t>::type dim0, size_t dim1,
         size_t dim2) : base(dim0, dim1, dim2) {}
 
+  // Default constructor added for compatibility with ComputeCPP and to pass
+  // SYCL conformance. Initializes range components to unit size.
+  range()
+      : base(detail::InitializedVal<dimensions,
+                                    detail::array>::template get<1>()) {}
```

Comment:
I'm not sure why ComputeCPP works that way, but I found it strange to initialize range with `1` by default.
I think current behavior (i.e. using `0`) is more logical as it indicates that the range wan't properly initialized using one of constructors defined by the specification.
+@Ruyk, @Naghasan to clarify why `1` is preferred here.

Reply:
> 1 is preferred here

Because that's the neutral for the multiplication. Although this is not specified in the specs as far as I remember. If so, you shouldn't have to do this in order to pass conformance.

Reply:
Yes, spec does not say anything, so conformance tests using the default constructor are invalid. But I think 1 is as good as 0, so it does not harm to change the default to what ComputeCpp uses. Updating the spec to use 1 as the default can be an OK fix too, the other one being fixing the tests.

Reply:
I think we should make 1 the value on the specification, simplifies a lot of the reasoning between ID's and Ranges.

Reply:
```c++
sycl::range<2> Size; // {1, 1}
sycl::id<2> Point; // {0, 0} or {1, 1}??
```

Reply:
id should start in 0, unless we plan to port SYCL to FORTRAN :-)